server/src/api/resolvers/people.resolver.ts

```diff
@@ -1,52 +1,72 @@
-import { PrismaClient } from '@prisma/client';
 import { Person } from '../generated-graphql/models';
 import { Resolver, Query, Args, Mutation } from '@nestjs/graphql';
 import { FindManyPersonArgs } from '../generated-graphql/resolvers/crud/Person/args/FindManyPersonArgs';
 import { UpdateOnePersonArgs } from '../generated-graphql/resolvers/crud/Person/args/UpdateOnePersonArgs';
 import { CreateOnePersonArgs } from '../generated-graphql/resolvers/crud/Person/args/CreateOnePersonArgs';
 import { DeleteManyPersonArgs } from '../generated-graphql/resolvers/crud/Person/args/DeleteManyPersonArgs';
 import { AffectedRowsOutput } from '../generated-graphql/resolvers/outputs/AffectedRowsOutput';
+import { UseGuards } from '@nestjs/common';
+import { JwtAuthGuard } from 'src/auth/guards/jwt.auth.guard';
+import { User, UserType } from './decorators/user.decorator';
+import { PrismaService } from 'src/database/prisma.service';
 
 @Resolver(() => Person)
 export class PeopleResolver {
-  constructor(private readonly prismaClient: PrismaClient) {}
+  constructor(private readonly prismaService: PrismaService) {}
 
+  @UseGuards(JwtAuthGuard)
   @Query(() => [Person], {
     nullable: false,
   })
   async people(@Args() args: FindManyPersonArgs): Promise<Person[]> {
-    return this.prismaClient.person.findMany({
+    return this.prismaService.person.findMany({
       ...args,
     });
   }
 
+  @UseGuards(JwtAuthGuard)
   @Mutation(() => Person, {
     nullable: true,
   })
   async updateOnePerson(
     @Args() args: UpdateOnePersonArgs,
   ): Promise<Person | null> {
-    return this.prismaClient.person.update({
+    if (!args.data.company?.connect?.id) {
+      delete args.data.company;
```

Comment:
Does this could mess with passed parameter ? As objects are passed by ref by default ?